src/nodes.coffee

```diff
@@ -515,9 +528,16 @@ exports.Root = class Root extends Base
     super()
 
     @isAsync = (new Code [], @body).isAsync
+    @referencedVars = new Set  # all referenced variable names in this scope
 
   children: ['body']
 
+  findReferencedVars: (scopeNodes = []) ->
+    # This is called at the top level to start the recursion,
```

Comment:
Is there any way to make this “automatic,” in the sense that we don’t need to remember to call `findReferencedVars` on the root, outside of `nodes.coffee`, to trigger this recursion? I assume we probably can’t call it in `Root`‘s constructor, as that would be too early? Could it happen within `compile` or `compileNode`, as maybe those get called after all the child nodes are added?